src/slurm_plugin/slurm_resources.py

```diff
@@ -107,22 +107,23 @@ def __eq__(self, other):
 
 class SlurmNode(metaclass=ABCMeta):
     SLURM_SCONTROL_BUSY_STATES = {"MIXED", "ALLOCATED", "COMPLETING"}
-    SLURM_SCONTROL_IDLE_STATE = "IDLE"
-    SLURM_SCONTROL_DOWN_STATE = "DOWN"
-    SLURM_SCONTROL_DRAIN_STATE = "DRAIN"
-    SLURM_SCONTROL_POWERING_DOWN_STATE = "POWERING_DOWN"
-    SLURM_SCONTROL_POWER_STATE = "IDLE+CLOUD+POWER"
-    SLURM_SCONTROL_POWER_UP_STATE = "#"
+    SLURM_SCONTROL_IDLE_STATE = {"IDLE"}
+    SLURM_SCONTROL_DOWN_STATES = {"DOWN"}
+    SLURM_SCONTROL_DRAIN_STATE = {"DRAIN"}
+    SLURM_SCONTROL_POWERING_DOWN_STATES = {"POWERING_DOWN", "POWER_DOWN"}
+    SLURM_SCONTROL_POWER_STATE = "IDLE+CLOUD+POWERED_DOWN"
+    SLURM_SCONTROL_POWER_UP_STATE = {"POWERING_UP"}
     SLURM_SCONTROL_ONLINE_STATES = {"IDLE+CLOUD", "MIXED+CLOUD", "ALLOCATED+CLOUD", "COMPLETING+CLOUD"}
-    SLURM_SCONTROL_POWER_WITH_JOB_STATE = "MIXED+CLOUD+POWER"
-    SLURM_SCONTROL_RESUME_FAILED_STATE = "DOWN*+CLOUD+POWER"
+    SLURM_SCONTROL_POWER_WITH_JOB_STATE = "MIXED+CLOUD+POWERED_DOWN"
+    SLURM_SCONTROL_RESUME_FAILED_STATE = "DOWN+CLOUD+POWERED_DOWN+NOT_RESPONDING"
 
     def __init__(self, name, nodeaddr, nodehostname, state, partitions=None, instance=None):
         """Initialize slurm node with attributes."""
         self.name = name
         self.nodeaddr = nodeaddr
         self.nodehostname = nodehostname
-        self.state = state
+        self.state_string = state
+        self.states = state.split("+")
```

Comment:
shall we force this to be a set and not a list otherwise you get O(n) and not O(1)

Reply:
Yes, change to self.states = set(state.split("+"))